### PR TITLE
feat: private CLI Rest infrastructure

### PR DESCRIPTION
**Pull request by rahulguptajss**
This PR introduces a new plugin "RestGaps". Example in volume.yaml.
Any rest gaps can be filled with relevant private cli call based on the instance key.

### CODE REVIEW

**Review by Hardikl — APPROVED** [review]

File: cmd/poller/plugin/restgaps/restgaps.go [hunk 1]
```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright NetApp Inc, 2021 All rights reserved
```
[the remark this answers is not in the document]
Comment (rahulguptajss) [reply cmd/poller/plugin/restgaps/restgaps.go:2]: looks like we'll need to update all of them. Will update them in 1 go later.

**Review by Hardikl — APPROVED** [review]

**Review by Hardikl — DISMISSED** [review]

**Comment by rahulguptajss** [comment]
ready for review.

**Review by cgrinds — APPROVED** [review]
Nice job @rahulguptajss the backwards compatibility testcases are especially nice!

File: pkg/tree/testdata/testTemplate21.08.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+name:                     Volume
+query:                    api/storage/volumes
```
Comment (Hardikl) [inline pkg/tree/testdata/testTemplate21.08.yaml:2]: zapi

**Pull request merged**